scheduler/core/meta/__init__.py

```diff
@@ -11,3 +14,13 @@ def __call__(cls, *args, **kwargs):
         if cls not in Singleton._instances:
             Singleton._instances[cls] = super().__call__(*args, **kwargs)
         return Singleton._instances[cls]
+
+
+@dataclass
+class AbstractDataclass(ABC):
+    pass
+
+
+@dataclass(frozen=True)
+class FrozenAbstractDataclass(ABC):
+    pass
```

Comment:
Not sure why we need these? Why wouldn't we just pass frozen or non-frozen dataclasses in? I could be missing something.

Reply:
The idea is to make `Interruption` and `Blockage` abstract. That could be done just using ABC on those classes too.

Reply:
Since this does nothing, we should probably get rid of these two and use the ABC route.